lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelAnalysis.cxx

```diff
@@ -255,6 +256,93 @@ Interval LinearModelAnalysis::getCoefficientsConfidenceInterval(const Scalar lev
   return bounds;
 }
 
+/* Check that the sample size is large enough to compute asymptotic distributions */
+void LinearModelAnalysis::checkSampleSize() const
+{
+  const UnsignedInteger size = linearModelResult_.getInputSample().getSize();
+  const UnsignedInteger minSize = ResourceMap::GetAsUnsignedInteger("LinearModelAnalysis-MinimumSampleSizeForAsymptoticDistributions");
+  if (!(size >= minSize))
+    throw InvalidArgumentException(HERE) << "The sample size is too small (must be at least " << minSize << ")";
+}
+
+SymmetricMatrix LinearModelAnalysis::getGramInverse() const
+{
+  return linearModelResult_.getDesign().computeGram().inverse();
+}
+
+/* Asymptotic distribution of the coefficients */
+Normal LinearModelAnalysis::getCoefficientsDistribution() const
+{
+  checkSampleSize();
+
+  const Point beta(linearModelResult_.getCoefficients());
+  const SymmetricMatrix gramInverse(getGramInverse());
+  const Scalar residualsVariance = linearModelResult_.getResidualsVariance();
+  const CovarianceMatrix covariance(residualsVariance * gramInverse);
+  return Normal(beta, covariance);
+}
+
+/* Asymptotic distribution of the variance of the residuals */
+Distribution LinearModelAnalysis::getVarianceDistribution(const Bool gaussian) const
+{
+  checkSampleSize();
+
+  const UnsignedInteger sampleSize = linearModelResult_.getInputSample().getSize();
+  const UnsignedInteger p = linearModelResult_.getCoefficients().getSize();
+  const Scalar residualsVariance = linearModelResult_.getResidualsVariance();
+  if (gaussian)
+  {
+    const Scalar k = (sampleSize - p) / 2.0;
+    const Scalar lambda = k / residualsVariance;
+    return Distribution(new Gamma(k, lambda));
+  }
+  else
+  {
+    const Scalar mu = residualsVariance;
+    const Scalar sd = residualsVariance * std::sqrt(2.0 * sampleSize) / (sampleSize - p);
+    return Distribution(new Normal(mu, sd));
+  }
+}
+
+Normal LinearModelAnalysis::computeDistributionForPredictionOrObservation(const Point & x0, const Bool observation) const
+{
+  checkSampleSize();
+
+  const Basis basis = linearModelResult_.getBasis();
```

Comment:
`const Basis basis(linearModelResult_.getBasis());` don't use copy constructor/affectation for complex types